app/Http/Controllers/Admin/RoleController.php

```diff
@@ -110,8 +110,21 @@ public function edit(Role $role): Application|Factory|View
      */
     public function update(RoleUpdateRequest $request, Role $role)
     {
-        if ($request->permissions) {
-            $role->syncPermissions($request->permissions);
+        $permissions = $request->permissions;
+
+        if ($permissions) {
+            // Do not add any other permissions if * is chosen.
```

Comment:
this bit of code does not execute when creating roles?

Reply:
if you want this to happen as well. i think you can update the RoleUpdateRequest and use the prepareForValidation() method to add this custom logic.